From d3de79dbfc20dc4dfc19154b16079861c542b71e Mon Sep 17 00:00:00 2001
From: Jiri Denemark <jdenemar@redhat.com>
Date: Tue, 2 Mar 2021 10:32:42 +0100
Subject: [PATCH] cpu_map: Install x86_EPYC-Milan.xml

Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
Reviewed-by: Pavel Hrdina <phrdina@redhat.com>

Origin: upstream, https://libvirt.org/git/?p=libvirt.git;a=commit;h=d3de79db
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1921880
Last-Update: 2021-04-16

---
 src/cpu_map/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cpu_map/meson.build b/src/cpu_map/meson.build
index 48f69f623c..013fc62a02 100644
--- a/src/cpu_map/meson.build
+++ b/src/cpu_map/meson.build
@@ -34,6 +34,7 @@ cpumap_data = [
   'x86_Dhyana.xml',
   'x86_EPYC-IBPB.xml',
   'x86_EPYC.xml',
+  'x86_EPYC-Milan.xml',
   'x86_EPYC-Rome.xml',
   'x86_features.xml',
   'x86_Haswell-IBRS.xml',
-- 
2.31.1

