Index: libvirt-0.8.3/src/util/bridge.c
===================================================================
--- libvirt-0.8.3.orig/src/util/bridge.c	2010-07-23 04:19:49.000000000 -0500
+++ libvirt-0.8.3/src/util/bridge.c	2010-08-11 16:59:54.000000000 -0500
@@ -244,7 +244,10 @@
                const char *bridge,
                const char *iface)
 {
-    return brAddDelInterface(ctl, SIOCBRADDIF, bridge, iface);
+    int ret;
+    if (ret = brAddDelInterface(ctl, SIOCBRADDIF, bridge, iface))
+        return ret;
+    return brSetInterfaceUp(ctl, bridge, 1);
 }
 # else
 int
@@ -613,7 +616,8 @@
         ifr.ifr_flags = flags;
 
         if (ioctl(ctl->fd, SIOCSIFFLAGS, &ifr) < 0)
-            return errno;
+            /*  Just smile and wave, boys... */
+            return 0;
     }
 
     return 0;
