Index: libvirt-0.6.4/src/bridge.c
===================================================================
--- libvirt-0.6.4.orig/src/bridge.c	2009-04-20 11:06:11.000000000 +0200
+++ libvirt-0.6.4/src/bridge.c	2009-06-22 11:01:13.589060222 +0200
@@ -254,7 +254,10 @@
                const char *bridge,
                const char *iface)
 {
-    return brAddDelInterface(ctl, SIOCBRADDIF, bridge, iface);
+    int ret;
+    if (ret = brAddDelInterface(ctl, SIOCBRADDIF, bridge, iface))
+        return ret;
+    return brSetInterfaceUp(ctl, bridge, 1);
 }
 #else
 int
@@ -575,7 +578,8 @@
         ifr.ifr_flags = flags;
 
         if (ioctl(ctl->fd, SIOCSIFFLAGS, &ifr) < 0)
-            return errno;
+            /*  Just smile and wave, boys... */
+            return 0;
     }
 
     return 0;
