Index: libvirt-0.8.5/src/util/bridge.c
===================================================================
--- libvirt-0.8.5.orig/src/util/bridge.c	2010-11-02 16:26:33.704307000 -0500
+++ libvirt-0.8.5/src/util/bridge.c	2010-11-02 16:29:19.163576003 -0500
@@ -245,7 +245,10 @@
                const char *bridge,
                const char *iface)
 {
-    return brAddDelInterface(ctl, SIOCBRADDIF, bridge, iface);
+    int ret;
+    if (ret = brAddDelInterface(ctl, SIOCBRADDIF, bridge, iface))
+        return ret;
+    return brSetInterfaceUp(ctl, bridge, 1);
 }
 # else
 int
@@ -616,7 +619,8 @@
         ifr.ifr_flags = flags;
 
         if (ioctl(ctl->fd, SIOCSIFFLAGS, &ifr) < 0)
-            return errno;
+            /*  Just smile and wave, boys... */
+            return 0;
     }
 
     return 0;
