Index: libvirt-0.6.4/src/virsh.c
===================================================================
--- libvirt-0.6.4.orig/src/virsh.c	2009-05-29 16:55:26.000000000 +0200
+++ libvirt-0.6.4/src/virsh.c	2009-06-22 11:03:18.268035966 +0200
@@ -57,6 +57,8 @@
 #define VSH_PROMPT_RW    "virsh # "
 #define VSH_PROMPT_RO    "virsh > "
 
+#include "remote_internal.h"
+
 #define GETTIMEOFDAY(T) gettimeofday(T, NULL)
 #define DIFF_MSEC(T, U) \
         ((((int) ((T)->tv_sec - (U)->tv_sec)) * 1000000.0 + \
@@ -7642,6 +7644,10 @@
 
     if ((defaultConn = getenv("VIRSH_DEFAULT_CONNECT_URI"))) {
         ctl->name = strdup(defaultConn);
+    } else if (!access(LIBVIRTD_PRIV_UNIX_SOCKET, W_OK)) {
+        ctl->name = strdup("qemu:///system");
+    } else {
+        ctl->name = strdup("qemu:///session");
     }
 
     if (!vshParseArgv(ctl, argc, argv)) {
@@ -7649,6 +7655,9 @@
         exit(EXIT_FAILURE);
     }
 
+    if (!ctl->quiet)
+        fprintf(stderr, _("Connecting to uri: %s\n"), ctl->name);
+
     if (!vshInit(ctl)) {
         vshDeinit(ctl);
         exit(EXIT_FAILURE);
