Index: libvirt-1.1.1/tools/Makefile.am
===================================================================
--- libvirt-1.1.1.orig/tools/Makefile.am	2014-03-25 14:15:10.818627185 -0500
+++ libvirt-1.1.1/tools/Makefile.am	2014-03-25 14:15:10.806627185 -0500
@@ -19,7 +19,9 @@ INCLUDES = \
 	-I$(top_builddir)/gnulib/lib -I$(top_srcdir)/gnulib/lib	\
 	-I$(top_builddir)/src -I$(top_srcdir)/src		\
 	-I$(top_srcdir)/src/util				\
+	-I$(top_srcdir)/src/remote				\
 	-I$(top_srcdir)						\
+	-DLOCAL_STATE_DIR=\""$(localstatedir)"\"                \
 	$(GETTEXT_CPPFLAGS)
 
 POD2MAN = pod2man -c "Virtualization Support" -r "$(PACKAGE)-$(VERSION)"
Index: libvirt-1.1.1/tools/Makefile.in
===================================================================
--- libvirt-1.1.1.orig/tools/Makefile.in	2014-03-25 14:15:10.818627185 -0500
+++ libvirt-1.1.1/tools/Makefile.in	2014-03-25 14:15:10.806627185 -0500
@@ -1909,7 +1909,9 @@ INCLUDES = \
 	-I$(top_builddir)/gnulib/lib -I$(top_srcdir)/gnulib/lib	\
 	-I$(top_builddir)/src -I$(top_srcdir)/src		\
 	-I$(top_srcdir)/src/util				\
+	-I$(top_srcdir)/src/remote				\
 	-I$(top_srcdir)						\
+	-DLOCAL_STATE_DIR=\""$(localstatedir)"\"                \
 	$(GETTEXT_CPPFLAGS)
 
 POD2MAN = pod2man -c "Virtualization Support" -r "$(PACKAGE)-$(VERSION)"
Index: libvirt-1.1.1/tools/virsh.c
===================================================================
--- libvirt-1.1.1.orig/tools/virsh.c	2014-03-25 14:15:10.818627185 -0500
+++ libvirt-1.1.1/tools/virsh.c	2014-03-25 14:15:10.810627185 -0500
@@ -84,6 +84,8 @@
 #include "virsh-snapshot.h"
 #include "virsh-volume.h"
 
+#include "remote_driver.h"
+
 static char *progname;
 
 static const vshCmdGrp cmdGroups[];
@@ -3189,14 +3191,18 @@ main(int argc, char **argv)
 
     if ((defaultConn = getenv("VIRSH_DEFAULT_CONNECT_URI"))) {
         ctl->name = vshStrdup(ctl, defaultConn);
-    }
+    } else if (!access(LIBVIRTD_PRIV_UNIX_SOCKET, W_OK)) {
+		ctl->name = vshStrdup(ctl, "qemu:///system");
+	} else {
+		ctl->name = vshStrdup(ctl, "qemu:///session");
+	}
 
-    if (!vshInit(ctl)) {
+    if (!vshParseArgv(ctl, argc, argv)) {
         vshDeinit(ctl);
         exit(EXIT_FAILURE);
     }
 
-    if (!vshParseArgv(ctl, argc, argv)) {
+    if (!vshInit(ctl)) {
         vshDeinit(ctl);
         exit(EXIT_FAILURE);
     }
