Index: libvirt-0.9.13/tools/Makefile.am
===================================================================
--- libvirt-0.9.13.orig/tools/Makefile.am	2012-06-26 20:44:39.000000000 -0500
+++ libvirt-0.9.13/tools/Makefile.am	2012-07-19 08:01:58.230054661 -0500
@@ -5,8 +5,10 @@
 	-I$(top_builddir)/include -I$(top_srcdir)/include	\
 	-I$(top_builddir)/gnulib/lib -I$(top_srcdir)/gnulib/lib	\
 	-I$(top_builddir)/src -I$(top_srcdir)/src		\
+	-I$(top_srcdir)/src/remote                      	\
 	-I$(top_srcdir)/src/util				\
 	-I$(top_srcdir)						\
+	-DLOCAL_STATE_DIR=\""$(localstatedir)"\"        	\
 	$(GETTEXT_CPPFLAGS)
 
 POD2MAN = pod2man -c "Virtualization Support" -r "$(PACKAGE)-$(VERSION)"
Index: libvirt-0.9.13/tools/Makefile.in
===================================================================
--- libvirt-0.9.13.orig/tools/Makefile.in	2012-07-01 22:26:48.000000000 -0500
+++ libvirt-0.9.13/tools/Makefile.in	2012-07-19 08:02:53.410053726 -0500
@@ -1509,8 +1509,10 @@
 	-I$(top_builddir)/include -I$(top_srcdir)/include	\
 	-I$(top_builddir)/gnulib/lib -I$(top_srcdir)/gnulib/lib	\
 	-I$(top_builddir)/src -I$(top_srcdir)/src		\
+	-I$(top_srcdir)/src/remote                        	\
 	-I$(top_srcdir)/src/util				\
 	-I$(top_srcdir)						\
+	-DLOCAL_STATE_DIR=\""$(localstatedir)"\"        	\
 	$(GETTEXT_CPPFLAGS)
 
 POD2MAN = pod2man -c "Virtualization Support" -r "$(PACKAGE)-$(VERSION)"
Index: libvirt-0.9.13/tools/virsh.c
===================================================================
--- libvirt-0.9.13.orig/tools/virsh.c	2012-07-19 07:50:13.000000000 -0500
+++ libvirt-0.9.13/tools/virsh.c	2012-07-19 08:04:16.134052326 -0500
@@ -67,6 +67,8 @@
 #include "intprops.h"
 #include "conf/virdomainlist.h"
 
+#include "remote_driver.h"
+
 static char *progname;
 
 #define VIRSH_MAX_XML_FILE 10*1024*1024
@@ -21050,7 +21052,11 @@
 
     if ((defaultConn = getenv("VIRSH_DEFAULT_CONNECT_URI"))) {
         ctl->name = vshStrdup(ctl, defaultConn);
-    }
+    } else if (!access(LIBVIRTD_PRIV_UNIX_SOCKET, W_OK)) {
+		ctl->name = vshStrdup(ctl, "qemu:///system");
+	} else {
+		ctl->name = vshStrdup(ctl, "qemu:///session");
+	}
 
     if (!vshParseArgv(ctl, argc, argv)) {
         vshDeinit(ctl);
