Index: ubuntu/src/virsh.c
===================================================================
--- ubuntu.orig/src/virsh.c	2009-11-27 11:15:39.324910902 +0100
+++ ubuntu/src/virsh.c	2009-11-27 21:20:24.639780576 +0100
@@ -57,6 +57,8 @@
 #define VSH_PROMPT_RW    "virsh # "
 #define VSH_PROMPT_RO    "virsh > "
 
+#include "remote_internal.h"
+
 #define GETTIMEOFDAY(T) gettimeofday(T, NULL)
 #define DIFF_MSEC(T, U) \
         ((((int) ((T)->tv_sec - (U)->tv_sec)) * 1000000.0 + \
@@ -8434,6 +8436,10 @@
 
     if ((defaultConn = getenv("VIRSH_DEFAULT_CONNECT_URI"))) {
         ctl->name = strdup(defaultConn);
+    } else if (!access(LIBVIRTD_PRIV_UNIX_SOCKET, W_OK)) {
+        ctl->name = strdup("qemu:///system");
+    } else {
+        ctl->name = strdup("qemu:///session");
     }
 
     if (!vshParseArgv(ctl, argc, argv)) {
