Index: libvirt-0.8.5/src/conf/capabilities.c
===================================================================
--- libvirt-0.8.5.orig/src/conf/capabilities.c	2010-11-02 16:24:03.203576001 -0500
+++ libvirt-0.8.5/src/conf/capabilities.c	2010-11-02 16:29:24.733576001 -0500
@@ -505,6 +505,13 @@
                                    const char *ostype)
 {
     int i;
+    /* Let's first see if we can match the host arch */
+    for (i = 0 ; i < caps->nguests ; i++) {
+        if (STREQ(caps->guests[i]->ostype, ostype) 
+              && STREQ(caps->guests[i]->arch.name, caps->host.arch))
+            return caps->guests[i]->arch.name;
+    }
+    /* If we couldn't we just grab the first match */
     for (i = 0 ; i < caps->nguests ; i++) {
         if (STREQ(caps->guests[i]->ostype, ostype))
             return 1;
