Description: match host arch as default
Author: Soren Hansen <soren@ubuntu.com>
Forwarded: no
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/344913

Index: libvirt-1.0.1/src/conf/capabilities.c
===================================================================
--- libvirt-1.0.1.orig/src/conf/capabilities.c	2013-01-22 10:43:03.812737360 -0600
+++ libvirt-1.0.1/src/conf/capabilities.c	2013-01-22 10:43:03.808737360 -0600
@@ -536,6 +536,13 @@
                                    const char *ostype)
 {
     int i;
+    /* Let's first see if we can match the host arch */
+    for (i = 0 ; i < caps->nguests ; i++) {
+        if (STREQ(caps->guests[i]->ostype, ostype) 
+              && STREQ(caps->guests[i]->arch.name, caps->host.arch))
+            return caps->guests[i]->arch.name;
+    }
+    /* If we couldn't we just grab the first match */
     for (i = 0 ; i < caps->nguests ; i++) {
         if (STREQ(caps->guests[i]->ostype, ostype))
             return 1;
