Index: libvirt-0.7.0/src/capabilities.c
===================================================================
--- libvirt-0.7.0.orig/src/capabilities.c	2009-08-05 07:50:00.000000000 -0500
+++ libvirt-0.7.0/src/capabilities.c	2009-08-13 20:37:30.000000000 -0500
@@ -478,6 +478,13 @@
                                    const char *ostype)
 {
     int i;
+    /* Let's first see if we can match the host arch */
+    for (i = 0 ; i < caps->nguests ; i++) {
+        if (STREQ(caps->guests[i]->ostype, ostype) 
+              && STREQ(caps->guests[i]->arch.name, caps->host.arch))
+            return caps->guests[i]->arch.name;
+    }
+    /* If we couldn't we just grab the first match */
     for (i = 0 ; i < caps->nguests ; i++) {
         if (STREQ(caps->guests[i]->ostype, ostype))
             return 1;
