--- libvirt-0.9.2.orig/configure
+++ libvirt-0.9.2/configure
@@ -54468,6 +54468,12 @@ $as_echo "yes" >&6; }
 fi
 fi
 
+# see https://bugs.launchpad.net/bugs/823711
+case "$host" in
+  arm*-*linux* )
+    LIBNL_LIBS=`echo $LIBNL_LIBS | sed 's/ *-lnl-[^ ]*//g'`
+    ;;
+esac
 
 
 
--- libvirt-0.9.2.orig/configure.ac
+++ libvirt-0.9.2/configure.ac
@@ -2382,6 +2382,12 @@ if test "$with_macvtap" = "yes"; then
     ])
 fi
 
+# see https://bugs.launchpad.net/bugs/823711
+case "$host" in
+  arm*-*linux* )
+    LIBNL_LIBS=`echo $LIBNL_LIBS | sed 's/ *-lnl-@<:@^ @:>@*//g'`
+    ;;
+esac
 AC_SUBST([LIBNL_CFLAGS])
 AC_SUBST([LIBNL_LIBS])
 
