From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 14 Jan 2012 18:09:01 +0100
Subject: Don't fail if we can't setup avahi
Index: libvirt-1.1.1/src/rpc/virnetserver.c
===================================================================
--- libvirt-1.1.1.orig/src/rpc/virnetserver.c	2014-03-25 14:15:10.470627177 -0500
+++ libvirt-1.1.1/src/rpc/virnetserver.c	2014-03-25 14:15:10.462627177 -0500
@@ -1055,8 +1055,7 @@ void virNetServerRun(virNetServerPtr srv
 
     virObjectLock(srv);
 
-    if (srv->mdns &&
-        virNetServerMDNSStart(srv->mdns) < 0)
+    if (srv->mdns) 
         goto cleanup;
 
     srv->quit = 0;
