From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 14 Jan 2012 18:09:01 +0100
Subject: Don't fail if we can't setup avahi

---
 src/rpc/virnetserver.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Index: libvirt-1.0.1/src/rpc/virnetserver.c
===================================================================
--- libvirt-1.0.1.orig/src/rpc/virnetserver.c	2013-01-22 10:43:03.208737371 -0600
+++ libvirt-1.0.1/src/rpc/virnetserver.c	2013-01-22 10:43:03.200737371 -0600
@@ -1079,8 +1079,7 @@
 
     virNetServerLock(srv);
 
-    if (srv->mdns &&
-        virNetServerMDNSStart(srv->mdns) < 0)
+    if (srv->mdns)
         goto cleanup;
 
     srv->quit = 0;
