Index: libvirt-1.1.1/src/qemu/qemu_capabilities.c
===================================================================
--- libvirt-1.1.1.orig/src/qemu/qemu_capabilities.c	2014-03-25 14:15:12.534627224 -0500
+++ libvirt-1.1.1/src/qemu/qemu_capabilities.c	2014-03-25 14:15:12.530627224 -0500
@@ -697,6 +697,7 @@ virQEMUCapsInitGuest(virCapsPtr caps,
     if (virQEMUCapsIsValidForKVM(hostarch, guestarch)) {
         const char *const kvmbins[] = { "/usr/libexec/qemu-kvm", /* RHEL */
                                         "qemu-kvm", /* Fedora */
+										"kvm-spice", /* qemu-kvm-spice Ubuntu package */
                                         "kvm" }; /* Upstream .spec */
 
         for (i = 0; i < ARRAY_CARDINALITY(kvmbins); ++i) {
