diff -Naurp libvirt-0.9.12.orig/daemon/Makefile.am libvirt-0.9.12/daemon/Makefile.am
--- libvirt-0.9.12.orig/daemon/Makefile.am	2012-04-20 00:25:25.000000000 -0400
+++ libvirt-0.9.12/daemon/Makefile.am	2012-05-16 10:59:58.242108358 -0400
@@ -95,7 +95,7 @@ libvirtd_SOURCES = $(DAEMON_SOURCES)
 libvirtd_CFLAGS = \
 	$(LIBXML_CFLAGS) $(GNUTLS_CFLAGS) $(SASL_CFLAGS) \
 	$(XDR_CFLAGS) $(POLKIT_CFLAGS) $(DBUS_CFLAGS) \
-	$(WARN_CFLAGS) \
+	$(WARN_CFLAGS) $(LIBNL_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-DQEMUD_PID_FILE="\"$(QEMUD_PID_FILE)\"" \
 	-DREMOTE_PID_FILE="\"$(REMOTE_PID_FILE)\""
diff -Naurp libvirt-0.9.12.orig/daemon/Makefile.in libvirt-0.9.12/daemon/Makefile.in
--- libvirt-0.9.12.orig/daemon/Makefile.in	2012-05-13 22:55:20.000000000 -0400
+++ libvirt-0.9.12/daemon/Makefile.in	2012-05-16 10:59:30.870107565 -0400
@@ -1565,7 +1565,7 @@ QEMU_PROTOCOL = $(top_srcdir)/src/remote
 @WITH_LIBVIRTD_TRUE@libvirtd_CFLAGS = \
 @WITH_LIBVIRTD_TRUE@	$(LIBXML_CFLAGS) $(GNUTLS_CFLAGS) $(SASL_CFLAGS) \
 @WITH_LIBVIRTD_TRUE@	$(XDR_CFLAGS) $(POLKIT_CFLAGS) $(DBUS_CFLAGS) \
-@WITH_LIBVIRTD_TRUE@	$(WARN_CFLAGS) \
+@WITH_LIBVIRTD_TRUE@	$(WARN_CFLAGS) $(LIBNL_CFLAGS) \
 @WITH_LIBVIRTD_TRUE@	$(COVERAGE_CFLAGS) \
 @WITH_LIBVIRTD_TRUE@	-DQEMUD_PID_FILE="\"$(QEMUD_PID_FILE)\"" \
 @WITH_LIBVIRTD_TRUE@	-DREMOTE_PID_FILE="\"$(REMOTE_PID_FILE)\""
